include/mips/tlb.h

```diff
@@ -61,6 +61,15 @@ void tlb_invalidate_all();
  * EntryLo0, EntryLo1, and parts in *hi, *lo0, *lo1 respectively. */
 void tlb_read_index(tlbhi_t *hi, tlblo_t *lo0, tlblo_t *lo1, unsigned idx);
 
+/* Reads the TLB entry specified by index and returns the EntryHi */
```

Comment:
Why so much trouble if you could define a simple wrapper and access statically allocated structure with gdb?

```c
static { tlbhi_t *hi, tlblo_t *lo0, tlblo_t *lo1 } _gdb_tlb_entry;

void _gdb_tlb_read_index(unsigned idx) {
  tlb_read_index(&_gdb_tlb_entry.hi, &_gdb_tlb_entry.lo0, &_gdb_tlb_entry.lo1, idx);
}
```